api/catalog/api/serializers/media_serializers.py

```diff
@@ -264,163 +167,263 @@ def validate(self, data):
+        return output
 
 
-class MediaSearchSerializer(serializers.Serializer):
-    """
-    This serializer serializes the full media search response. The class should
-    be inherited by all individual media serializers.
-    """
+#######################
+# Dynamic serializers #
+#######################
 
-    result_count = serializers.IntegerField(
-        help_text="The total number of items returned by search result."
-    )
-    page_count = serializers.IntegerField(
-        help_text="The total number of pages returned by search result."
-    )
-    page_size = serializers.IntegerField(help_text="The number of items per page.")
-    page = serializers.IntegerField(
-        help_text="The current page number returned in the response."
-    )
 
+def get_search_request_source_serializer(media_type):
+    class MediaSearchRequestSourceSerializer(serializers.Serializer):
+        """
+        This serializer parses and validates the source/not_source fields from the query
+        parameters.
+        """
 
-class MediaThumbnailRequestSerializer(serializers.Serializer):
-    """
-    This serializer parses and validates thumbnail query string parameters.
-    """
+        field_names = [
+            "source",
+            "excluded_source",
+        ]
+        """
+        Keep the fields names in sync with the actual fields below as this list is
+        used to generate Swagger documentation.
+        """
 
-    full_size = serializers.BooleanField(
-        source="is_full_size",
-        allow_null=True,
-        required=False,
-        default=False,
-        help_text="whether to render the actual image and not a thumbnail version",
-    )
-    compressed = serializers.BooleanField(
-        source="is_compressed",
-        allow_null=True,
-        default=None,
-        required=False,
-        help_text="whether to compress the output image to reduce file size,"
-        "defaults to opposite of `full_size`",
-    )
+        _field_attrs = {
+            "help_text": make_comma_separated_help_text(
+                search_controller.get_sources(media_type).keys(), "data sources"
+            ),
+            "required": False,
+        }
 
-    def validate(self, data):
-        if data.get("is_compressed") is None:
-            data["is_compressed"] = not data["is_full_size"]
-        return data
+        source = serializers.CharField(
+            label="provider",
+            **_field_attrs,
+        )
+        excluded_source = serializers.CharField(
+            label="excluded_provider",
+            **_field_attrs,
+        )
+
+        @staticmethod
+        def validate_source_field(input_sources):
+            allowed_sources = list(search_controller.get_sources(media_type).keys())
+            input_sources = input_sources.split(",")
+            input_sources = [x for x in input_sources if x in allowed_sources]
+            input_sources = ",".join(input_sources)
+            return input_sources.lower()
```

Comment:
Do we want to perform this `lower` step before any comparisons are made to normalize input vs allowed?

Reply:
This is good, addressed in 3d4f26b.